lib/Dialect/Openfhe/Transforms/ConfigureCryptoContext.cpp

```diff
@@ -0,0 +1,161 @@
+#include "lib/Dialect/Openfhe/Transforms/ConfigureCryptoContext.h"
+
+#include <cstddef>
+#include <string>
+
+#include "lib/Dialect/Openfhe/IR/OpenfheOps.h"
+#include "lib/Dialect/Openfhe/IR/OpenfheTypes.h"
+#include "mlir/include/mlir/Dialect/Arith/IR/Arith.h"   // from @llvm-project
+#include "mlir/include/mlir/Dialect/Func/IR/FuncOps.h"  // from @llvm-project
+#include "mlir/include/mlir/IR/ImplicitLocOpBuilder.h"  // from @llvm-project
+
+namespace mlir {
+namespace heir {
+namespace openfhe {
+
+#define GEN_PASS_DEF_CONFIGURECRYPTOCONTEXT
+#include "lib/Dialect/Openfhe/Transforms/Passes.h.inc"
+
+// Helper function to check if the function has MulOp or MulNoRelinOp
+bool hasMulOp(func::FuncOp op) {
+  bool result = false;
+  op.walk<WalkOrder::PreOrder>([&](Operation *op) {
+    if (isa<openfhe::MulOp>(op) || isa<openfhe::MulNoRelinOp>(op) ||
+        isa<openfhe::MulPlainOp>(op)) {
+      result = true;
+      return WalkResult::interrupt();
+    }
+    return WalkResult::advance();
+  });
+  return result;
+}
+
+// Helper function to find all the rotation indices in the function
+SmallVector<int64_t> findAllRotIndices(func::FuncOp op) {
+  SmallVector<int64_t> rotIndices;
+  op.walk([&](openfhe::RotOp rotOp) {
+    // TODO : I'm not sure if this is the right way to get the index
```

Comment:
I think that seems fine with me. I think though the rotation index is always going to be statically known, and I wonder whether we should convert this to an integer attribute instead of an SSA value (not in this PR)... @j2kun wdyt?

Reply:
I'll also add this to the issue thread.